### PR TITLE
Created domain model for all user stories

### PR DESCRIPTION
**Pull request by 04tudor**

### CODE REVIEW

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import md.maib.retail.Model.Conditions.Rule;
+import md.maib.retail.Model.Effects.LoyaltyEventType;
+import org.threeten.extra.Interval;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+@Getter
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java]: Add empty line after imports

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@Setter
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java]: We don't need setters. In domain-driven design we create business methods for changing domain entities

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@NoArgsConstructor
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java]: We will create campaigns with all needed fields, so we don't need NoArgsConstructor

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignId.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package md.maib.retail.Model.Campaign;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import java.util.Objects;
+import java.util.UUID;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+@ToString
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignId.java]: Add empty line

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java (hunk 1 above)
Comment (04tudor) [reply domain-model/src/main/java/md/maib/retail/Model/Campaign/Campaign.java]: After I've done it, do I need to create a new commit and push?
or edit it only in IDE

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignId.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+@Getter
+@Setter
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignId.java]: CampaignId is immutable object so we don't need Setters

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignMetaInfo.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.ToString;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+@ToString
+public class CampaignMetaInfo {
+
```
Comment (DemiusAcademius) [inline domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignMetaInfo.java]: record automatically realize toString, equals & hashCode methods

public record CampaignMetaInfo(Map<String, Object> properties) {

    public CampaignMetaInfo {
        notNull(properties, "properties");
    }
}

**Review by DemiusAcademius — CHANGES_REQUESTED** [review]

File: domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignMetaInfo.java (hunk 6 above)
Comment (04tudor) [reply domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignMetaInfo.java]: like this?
package md.maib.retail.Model.Campaign;


import java.util.Map;
import static java.util.Objects.requireNonNull;

public record CampaignMetaInfo(Map<String, Object> properties) {

    public CampaignMetaInfo{
        requireNonNull(properties, "Properties must not be null.");    }
}

i should to make and CampaignId as a record ?

**Pull request merged**